nuxeo-opencmis-impl/src/main/java/org/nuxeo/ecm/core/opencmis/impl/server/CMISQLtoNXQL.java

```diff
@@ -430,6 +435,19 @@ protected String getColumn(String propertyId) {
         if (propertyId.startsWith(CMIS_PREFIX) || propertyId.startsWith(NX_PREFIX)) {
             return getSystemColumn(propertyId);
         } else {
+            if (useElasticSearch && propertyId.indexOf(":") == -1) {
```

Comment:
But is the `useElasticSearch` boolean really needed though? Your fix adds a prefix when it's missing, and all other NXQL Core backends (VCS, DBS) already allow prefixed versions for properties of non-prefixed schemas, as tested in `TestSQLRepositoryQuery.testQueryBasic` (forget the "SQL" part in the class name, this is tested for all backends). So I think you can leave that out and simplify the code.

Reply:
I thought about this.  There's a cost to looping through the schemas for each property.  As the non-elastic backends currently worked, I thought it was probably best to just incur that cost for the jira I was solving - rather than changing the behaviour for all backends.  I can change it if you like.

Reply:
It's good to think about these kinds of costs. However in this case it's still small and we'll want to refactor this to precompute everything and move it to the SchemaManager at some point anyway.